### PR TITLE
Pull in cargo-xbuild as a dep

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cargo-n64/src/cargo.rs [hunk 1]
```diff
@@ -83,7 +84,7 @@ crate fn run(args: &cli::BuildArgs) -> Result<CargoArtifact, SubcommandError> {
         args
     };
 
-    let output = Command::new("cargo")
+    let output = Command::new(env::current_exe().expect("Couldn't get cargo-n64 executable path"))
```
Comment (parasyte) [inline cargo-n64/src/cargo.rs]: It might be better to map the `Error` case to a new `SubcommandError` variant.

File: cargo-n64/Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ goblin = { version = "0.2", default-features = false, features = ["std", "elf32"
 itertools = "0.8"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
+cargo-xbuild = "=0.5.25"
```
Comment (parasyte) [inline cargo-n64/Cargo.toml]: I'm happy with a loose semver, here.

Even though a breaking change ended up in a patch release last time. But that's ok! It happens. That's what the CI cron job is for.

Also, this is really nitpicky, but would you mind maintaining alphabetical sorting in this list?

File: cargo-n64/src/lib.rs [hunk 3]
```diff
@@ -148,6 +149,17 @@ where
 pub fn run() -> Result<(), RunError> {
     use self::RunError::*;
 
+    let args = env::args().collect::<Vec<_>>();
+
+    // So users won't have to install an extra cargo command and worry about its version being
+    // up to date, we have cargo-xbuild as a dep, and just transfer control to it when we're being
+    // invoked as such.
```
Comment (parasyte) [inline cargo-n64/src/lib.rs:156]: Clever girl.

**Review by parasyte — CHANGES_REQUESTED** [review]
Thank you do much for this! I was just today thinking about sitting down and doing the work. I even created a tracking ticket to remind myself before I noticed the PR. You beat me to it by several hours. 😅

I have just one change request; try not to panic.

File: cargo-n64/src/cargo.rs (hunk 1 above)
Comment (ghost) [reply cargo-n64/src/cargo.rs]: New variant added.

File: cargo-n64/Cargo.toml (hunk 2 above)
Comment (ghost) [reply cargo-n64/Cargo.toml]: I'm a stickler for alphabetical sorting too. 😁 Semver loosened and deps sorted.

**Review by parasyte — APPROVED** [review]

**Comment by parasyte** [comment]
Wow! This adds almost 1MB to bin size.

Before:

```
$ ls -l target/release/cargo-n64
-rwxr-xr-x  2 parasyte  staff  909096 Feb 20 19:33 target/release/cargo-n64*

$ cargo bloat --release -p cargo-n64
   Compiling cargo-n64 v0.1.2 (/Users/parasyte/projects/cargo-n64/cargo-n64)
    Finished release [optimized] target(s) in 6.05s
    Analyzing target/release/cargo-n64

 File  .text     Size          Crate Name
 1.6%   2.7%  14.5KiB            std std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::spawn
 1.2%   2.0%  10.6KiB      [Unknown] _read_line_info
 1.2%   1.9%  10.2KiB      [Unknown] _read_line_info
 1.1%   1.9%  10.0KiB         goblin goblin::elf::Elf::parse
 0.9%   1.5%   8.0KiB      cargo_n64 cargo_n64::run
 0.9%   1.4%   7.6KiB  backtrace_sys ___rbt_backtrace_dwarf_add
 0.8%   1.3%   7.0KiB  backtrace_sys ___rdos_backtrace_dwarf_add
 0.8%   1.3%   6.8KiB     serde_json <serde_json::read::StrRead as serde_json::read::Read>::parse_str
 0.7%   1.2%   6.2KiB      cargo_n64 cargo_n64::cli::parse_args
 0.6%   1.1%   5.8KiB         goblin goblin::elf::dynamic::Dynamic::parse
 0.6%   1.0%   5.3KiB      [Unknown] __mh_execute_header
 0.6%   1.0%   5.1KiB      [Unknown] _read_attribute
 0.6%   0.9%   5.0KiB            std core::num::flt2dec::strategy::dragon::format_shortest
 0.5%   0.9%   4.7KiB      [Unknown] _read_attribute
 0.5%   0.9%   4.6KiB rustc_demangle rustc_demangle::v0::Printer::print_type
 0.5%   0.8%   4.5KiB            std alloc::collections::btree::map::BTreeMap<K,V>::insert
 0.5%   0.8%   4.5KiB rustc_demangle rustc_demangle::v0::Printer::print_type
 0.5%   0.8%   4.5KiB      cargo_n64 cargo_n64::cli::parse_build_args
 0.5%   0.8%   4.2KiB rustc_demangle <rustc_demangle::legacy::Demangle as core::fmt::Display>::fmt
 0.5%   0.8%   4.2KiB rustc_demangle <rustc_demangle::legacy::Demangle as core::fmt::Display>::fmt
46.2%  76.5% 410.1KiB                And 1074 smaller methods. Use -n N to show more.
60.4% 100.0% 536.0KiB                .text section size, the file size is 887.8KiB
```

After:

```
$ ls -l target/release/cargo-n64
-rwxr-xr-x  2 parasyte  staff  1705384 Feb 20 19:33 target/release/cargo-n64*

$ cargo bloat --release -p cargo-n64
    Finished release [optimized] target(s) in 9.21s
    Analyzing target/release/cargo-n64

 File  .text     Size           Crate Name
 1.0%   1.6%  17.3KiB            toml toml::parser::Parser::value
 0.9%   1.4%  15.4KiB            toml toml::parser::Parser::number_or_datetime
 0.9%   1.4%  15.2KiB            toml toml::parser::Parser::parse
 0.9%   1.3%  14.5KiB             std std::sys::unix::process::process_inner::<impl std::sys::unix::process::process_common::Command>::spawn
 0.8%   1.2%  13.5KiB cargo_metadata? <cargo_metadata::_IMPL_DESERIALIZE_FOR_Package::<impl serde::de::Deserialize for cargo_metadata::Package>::deserialize::__Visitor as serde::de::Visitor>::visit_map
 0.8%   1.2%  13.0KiB       xargo_lib xargo_lib::sysroot::build
 0.7%   1.0%  11.4KiB  cargo_metadata <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
 0.6%   1.0%  10.6KiB       [Unknown] _read_line_info
 0.6%   0.9%  10.2KiB       [Unknown] _read_line_info
 0.6%   0.9%  10.0KiB          goblin goblin::elf::Elf::parse
 0.6%   0.9%   9.7KiB  cargo_metadata <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
 0.5%   0.8%   8.9KiB  cargo_metadata <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
 0.5%   0.8%   8.4KiB       xargo_lib xargo_lib::sysroot::update
 0.5%   0.8%   8.4KiB       cargo_n64 cargo_n64::run
 0.5%   0.7%   7.6KiB   backtrace_sys ___rbt_backtrace_dwarf_add
 0.4%   0.7%   7.4KiB  cargo_metadata <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
 0.4%   0.7%   7.3KiB            toml toml::parser::Parser::finish_basic_string
 0.4%   0.6%   7.0KiB   backtrace_sys ___rdos_backtrace_dwarf_add
 0.4%   0.6%   6.8KiB      serde_json <serde_json::read::StrRead as serde_json::read::Read>::parse_str
 0.4%   0.6%   6.5KiB  cargo_metadata <&mut serde_json::de::Deserializer<R> as serde::de::Deserializer>::deserialize_struct
53.5%  81.3% 891.5KiB                 And 2127 smaller methods. Use -n N to show more.
65.8% 100.0%   1.1MiB                 .text section size, the file size is 1.6MiB
```

`toml` and `cargo_metadata` are both huge. I was hoping there was some quick and dirty feature flags we could enable to reduce binary size and compile time. Maybe some sub-crates can be made smaller? Here is the list of new transitive dependencies:

```
cargo-xbuild v0.5.25
├── cargo_metadata v0.9.1
│   ├── semver v0.9.0
│   │   ├── semver-parser v0.7.0
│   │   └── serde v1.0.104
│   │       └── serde_derive v1.0.104
│   │           ├── proc-macro2 v1.0.8
│   │           │   └── unicode-xid v0.2.0
│   │           ├── quote v1.0.2
│   │           │   └── proc-macro2 v1.0.8 (*)
│   │           └── syn v1.0.14
│   │               ├── proc-macro2 v1.0.8 (*)
│   │               ├── quote v1.0.2 (*)
│   │               └── unicode-xid v0.2.0 (*)
│   ├── serde v1.0.104 (*)
│   ├── serde_derive v1.0.104 (*)
│   └── serde_json v1.0.48
│       ├── itoa v0.4.5
│       ├── ryu v1.0.2
│       └── serde v1.0.104 (*)
├── error-chain v0.7.2
├── fs2 v0.4.3
│   └── libc v0.2.66
├── libc v0.2.66 (*)
├── rustc_version v0.1.7
│   └── semver v0.1.20
├── serde v1.0.104 (*)
├── serde_derive v1.0.104 (*)
├── serde_json v1.0.48 (*)
├── tempdir v0.3.7
│   ├── rand v0.4.6
│   │   └── libc v0.2.66 (*)
│   └── remove_dir_all v0.5.2
├── toml v0.2.1
│   └── rustc-serialize v0.3.24
└── walkdir v1.0.7
    └── same-file v0.1.3
```

**Pull request merged**